packages/grafana-ui/src/components/Forms/Select/Select.story.tsx

```diff
@@ -247,17 +247,23 @@ export const customizedControl = () => {
 
 export const customValueCreation = () => {
   const [value, setValue] = useState<SelectableValue<string>>();
-
+  const [customOptions, setCustomOptions] = useState<Array<SelectableValue<string>>>([]);
+  const options = generateOptions();
   return (
     <>
       <Select
-        options={generateOptions()}
+        options={[...options, ...customOptions]}
         value={value}
         onChange={v => {
           setValue(v);
         }}
         size="md"
         allowCustomValue
+        onCreateOption={v => {
+          const customValue: SelectableValue<string> = { value: kebabCase(v), label: v };
```

Comment:
Why kebabCase on the value?

Reply:
ah, that's just for storybook, to to keep it consistent with the generated options. In actual implementation this would be most probably some `id` created by backend service.